File: ios/CounterViewManager.m
//
//  CounterViewManager.m
//  CounterApp
//
//  Created by 黃立洋 on 2022/10/9.
//

#import <Foundation/Foundation.h>
#import "React/RCTViewManager.h"
@interface RCT_EXTERN_MODULE(CounterViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(count, NSNumber)
RCT_EXPORT_VIEW_PROPERTY(onUpdate, RCTDirectEventBlock)
RCT_EXTERN_METHOD(
  updateFromManager:(nonnull NSNumber *)node
  count:(nonnull NSNumber *)count
)
@end

File: ios/CounterViewManager.swift
//
//  CounterViewManager.swift
//  CounterApp
//
//  Created by 黃立洋 on 2022/10/9.
//

import Foundation


@objc(CounterViewManager)
class CounterViewManager: RCTViewManager {
  override func view() -> UIView! {
//    let label = UILabel()
//    label.text = "Swift Counter"
//    label.textAlignment = .center
//    return label
    return CounterView()
  }
  override static func requiresMainQueueSetup() -> Bool {
      return true
  }
  // 1
  @objc func updateFromManager(_ node: NSNumber, count: NSNumber) {
      
      DispatchQueue.main.async {                                // 2
        let component = self.bridge.uiManager.view(             // 3
          forReactTag: node                                     // 4
        ) as! CounterView                                       // 5
        component.update(value: count)                          // 6
      }
    }
}

